### PR TITLE
AVRO-3686: Update the website on changes in doc/ folder in master branch

### PR DESCRIPTION
**Pull request by martin-g**
AVRO-3686

## What is the purpose of the change

* Update the [website](https://avro.apache.org) on update to the [doc](https://github.com/apache/avro/tree/master/doc) folder contents.

Workflow run could be seen at https://github.com/apache/avro/actions/runs/4688248904

## Verifying this change

See https://avro.apache.org/update-site/
It has some broken links, e.g. the logo and the links to the APIs, because they break when a custom baseUrl (`/update-site`) is used.

## Documentation

- Does this pull request introduce a new feature? no

### CODE REVIEW

**Comment by martin-g** [comment]
Compared to the current/old version of the website the new one shows the content of `++version++` as the main one, i.e. the non-released version.
The users of 1.11 will have to navigate to Documentation -> 1.11.1.

**Review by opwvhk — APPROVED** [review]

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        run: |
+          set -x
+          cd lang/java
+          mvn javadoc::aggregate
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: api-java
+          path: lang/java/target/site/apidocs
+
```
Comment (kojiromike) [inline .github/workflows/docs.yaml:163]:
```suggestion

  build-api-python:
    name: Build Python API docs
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: Setup Python
        uses: actions/setup-python@v4
        with:
          python-version: 3.11

      - name: Install tox
        run: python3 -m pip install tox
        
      - name: Build docs
        working-directory: lang/py
        run: ./build.sh doc

      - uses: actions/upload-artifact@v3
        with:
          name: api-python
          path: lang/py/target/site/apidocs
```

File: .github/workflows/docs.yaml [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        with:
+          name: api-csharp
+          path: api-csharp
+
+      - name: Download api-java
+        uses: actions/download-artifact@v3
+        with:
+          name: api-java
+          path: api-java
+
```
Comment (kojiromike) [inline .github/workflows/docs.yaml:204]:
```suggestion

      - name: Download api-python
        uses: actions/download-artifact@v3
        with:
          name: api-python
          path: api-python
```

File: .github/workflows/docs.yaml [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+          mkdir -p website/docs/++version++/api/c
+          mkdir -p website/docs/++version++/api/cpp/html
+          mkdir -p website/docs/++version++/api/csharp/html
+          mkdir -p website/docs/++version++/api/java
+
+          mv api-c/* website/docs/++version++/api/c/
+          mv api-c++/* website/docs/++version++/api/cpp/html/
+          mv api-csharp/* website/docs/++version++/api/csharp/html/
+          mv api-java/* website/docs/++version++/api/java/
```
Comment (kojiromike) [inline .github/workflows/docs.yaml:217]:
```suggestion
          mv api-java/* website/docs/++version++/api/java/
          mv api-python/* website/docs/++version++/api/python/html/
```

File: .github/workflows/docs.yaml [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - name: Install NPM dependencies
+        run: |
+          set -x
+          cd doc
+          npm install
```
Comment (kojiromike) [inline .github/workflows/docs.yaml:50]:
```suggestion
        working-directory: doc
        run: npm install
```

File: .github/workflows/docs.yaml [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+        uses: peaceiris/actions-hugo@v2
+        with:
+          hugo-version: 0.111.3
+          extended: true
+
+      - name: Build docs
+        run: |
+          set -x
+          cd doc
+          hugo --minify --destination ../website --baseURL=/
```
Comment (kojiromike) [inline .github/workflows/docs.yaml:62]:
```suggestion
        working-directory: doc
        run: hugo --minify --destination ../website --baseURL=/
```

**Comment by martin-g** [comment]
Done with https://github.com/apache/avro/pull/3112

**Pull request closed without merging**